src/maintenance/services/coordinate_arduino_review.py

```diff
@@ -0,0 +1,141 @@
+"""
+Monitor status of specific Redis locks and communicate it to Arduino.
+
+Run with supervisor, as it doesn't daemonize or check for its health.
+
+Usage:
+    coordinate-arduino [-d DEVICE] [-s DEVICES] [-f] LOCKS...
+    coordinate-arduino -h | --help
+    coordinate-arduino --version
+
+LOCKS are an ordered list of Redis keys to poll.
+
+Options:
+    -f          Flatten all locks to the least-significant bit.
+    -d DEVICE   Device to communicate with Arduino [default: /dev/ttyACM0].
+    -s DEVICES  Scan for available devices with glob patterns.
+    -h, --help  Display this message.
+    --version   Show version information.
+
+"""
+
+VERSION = 1.0
+
+
+from docopt import docopt
+import serial
+import time
+import redis
+
+import glob
+
+import time
+
+from pathlib import Path
+
+from ..config.coordinate import CoordinateConfigFile
+from ..arduino_api import Message
+
+from ..strings import _regex_splitter
+
+from itertools import chain, repeat
+from functools import partial, reduce
+
+from ..functional import compose
+
+
+def find_suitable_device(device, patterns):
+    p = Path(device)
+
+    if p.exists():
+        return p
+
+    suitable_paths = chain(*(glob.glob(pattern) for pattern in patterns))
+
+    try:
+        return next(map(lambda x: Path(x), suitable_paths))
+    except StopIteration:
+        raise FileNotFoundError("There is no device under {} and {}".format(device, ', '.join(patterns)))
```

Comment:
W tej funkcji dzieje się nieco za dużo: najpierw sprawdzasz czy device istnieje a jak nie to przeglądasz globy. Generalnie ja bym to rozbił na dwie funkcje, każdą przyjmującą po jednym parametrze i wołał je potem jakoś w stylu:

```python
def configured_find_suitable_device():
    return get_device_by_path(path) or get_device_by_patterns(patterns)
```

Prościej, czytelniej i nie naciągamy SRP ;)